fw-child/resources/js/map.js

```diff
@@ -1350,39 +1351,52 @@
         options = plugin.options,
         item = plugin.item;
 
-      // todo: implement behaviour for building_climate_zones
-      let layer_name = $(document).cdc_app(
-        'maps.get_layer_name',
-        options.query,
+      let default_scheme_element = item.find(
+        '#display-scheme-select .dropdown-item[data-scheme-id="default"]',
       );
-      $.getJSON(
-        geoserver_url +
+
+      if (special_variables.hasOwnProperty(options.var_data.slug)) {
+        default_scheme_element.data('scheme-colours', special_variables[options.var_data.slug].colormap.colours);
+        default_scheme_element.data('scheme-quantities', special_variables[options.var_data.slug].colormap.quantities);
+        plugin.update_scheme();
+
+        if (typeof callback == 'function') {
+          callback();
+        }
+      } else {
+
+        // todo: implement behaviour for building_climate_zones
+        let layer_name = $(document).cdc_app(
+          'maps.get_layer_name',
+          options.query,
+        );
+        $.getJSON(
+          geoserver_url +
           '/geoserver/wms?service=WMS&version=1.1.0&request=GetLegendGraphic' +
           '&format=application/json&layer=' +
           layer_name,
-      )
-        .then(function (data) {
-          let colour_map =
-            data.Legend[0].rules[0].symbolizers[0].Raster.colormap.entries;
+        )
+          .then(function (data) {
+            let colour_map =
+              data.Legend[0].rules[0].symbolizers[0].Raster.colormap.entries;
 
-          let default_scheme_element = item.find(
-            '#display-scheme-select .dropdown-item[data-scheme-id="default"]',
-          );
-          default_scheme_element.data(
-            'scheme-colours',
-            colour_map.map((e) => e.color),
-          );
-          default_scheme_element.data(
-            'scheme-quantities',
-            colour_map.map((e) => parseFloat(e.quantity)),
-          );
-          plugin.update_scheme();
-        })
-        .always(function () {
-          if (typeof callback == 'function') {
-            callback();
-          }
-        });
+
```

Comment:
As per the _[boy scout rule](https://deviq.com/principles/boy-scout-rule)_, maybe we should remove the extra empty lines here.

Reply:
I lost track of that line... is it good now?